### PR TITLE
don't leak file descriptors <3

### PR DESCRIPTION
**Pull request by mahkoh**
Don't merge before old_io has been removed.

cc @alexcrichton @aturon

### CODE REVIEW

**Comment by mahkoh** [comment]
Fixes https://github.com/rust-lang/rust/issues/19028

File: src/libstd/sys/unix/fd.rs [hunk 1]
```diff
@@ -60,14 +60,10 @@ impl AsInner<c_int> for FileDesc {
...
-        // file descriptor was closed or not, and if we retried (for something
-        // like EINTR), we might close another valid file descriptor (opened
-        // after we closed ours.
-        if self.fd > libc::STDERR_FILENO {
-            let _ = unsafe { libc::close(self.fd) };
-        }
+        // Note that errors are ignored when closing a file descriptor. The reason for
+        // this is that if an error occurs we don't actually know if the file descriptor
+        // was closed or not, and if we retried (for something like EINTR), we might close
+        // another valid file descriptor (opened after we closed ours.
```
Comment (huonw) [inline src/libstd/sys/unix/fd.rs]: Could you add the missing ) while you're here?

**Comment by alexcrichton** [comment]
While I agree that we should land this soon, I do not believe that we can do so currently due to the reliance of `old_io`. I believe, for example, that the test suite does not pass if this is landed.

In the interest of keeping the queue relatively clear I'm going to close this for now but we can certainly land once `old_io` is dealt with.

**Comment by l0kod** [comment]
I hope #22797 can help to merge this PR.

**Comment by l0kod** [comment]
cc #12148

**Pull request closed without merging**